builtin/providers/heroku/resource_heroku_space.go

```diff
@@ -0,0 +1,111 @@
+package heroku
+
+import (
+	"context"
+	"log"
+
+	heroku "github.com/cyberdelia/heroku-go/v3"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceHerokuSpace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceHerokuSpaceCreate,
+		Read:   resourceHerokuSpaceRead,
+		Update: resourceHerokuSpaceUpdate,
+		Delete: resourceHerokuSpaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"organization": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceHerokuSpaceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*heroku.Service)
+
+	opts := heroku.SpaceCreateOpts{}
+	opts.Name = d.Get("name").(string)
+	opts.Organization = d.Get("organization").(string)
+
+	if v, ok := d.GetOk("region"); ok {
+		vs := v.(string)
+		opts.Region = &vs
+	}
+
+	space, err := client.SpaceCreate(context.TODO(), opts)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(space.ID)
+	log.Printf("[INFO] Space ID: %s", d.Id())
+
+	setSpaceAttributes(d, (*heroku.Space)(space))
```

Comment:
If the vendored version of heroku-go is updated, this type conversion can be dropped, as the `Space` methods have been updated to all return `*heroku.Space`.